### PR TITLE
chore(tracing/integrations)!: remove deprecated manual instrumentation helpers

### PR DESCRIPTION
**Pull request by mabdinur**
Removes:
 - ddtrace.contrib.flask.middleware.py:TraceMiddleware()
 - ddtrace.contrib.pymongo.patch.py:pytrace_mongo_client()
 - ddtrace.contrib.pymysql.tracers.py:get_traced_pymysql_connection()
 - ddtrace.contrib.redis.tracers.py:get_traced_redis()
 - ddtrace.contrib.redis.tracers.py:get_traced_redis_from()

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
why are we moving these to 2.0.0?

shouldn't these be removed now?

**Comment by mabdinur** [comment]
> why are we moving these to 2.0.0?
> 
> shouldn't these be removed now?

I believe we pushed updating integrations to v2.0. Is this correct @Kyle-Verhoog @majorgreys?

**Comment by mabdinur** [comment]
Made this change in `0.x`: https://github.com/DataDog/dd-trace-py/pull/3278/commits/f9b2126127d221bc3e118ad1849f79f6c03a458e

File: releasenotes/notes/update-deprecated-contrib-removal-version-ca920d9e1d91b6e0.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - |
+    Removed deprecated middleware ``ddtrace.contrib.flask.middleware.py:TraceMiddleware()``. Use flask patching instead.
+  - |
+    Removed deprecated function ``ddtrace.contrib.pymongo.patch.py:trace_mongo_client()``. Use pymongo patching instead.
+  - |
+    Removed deprecated function ``ddtrace.contrib.pymysql.tracers.py:get_traced_pymysql_connection()``. Use pymysql patching instead.
+  - |
+    Removed deprecated function ``ddtrace.contrib.redis.tracers.py:get_traced_redis()`` helper. Use redis patching instead.
+  - |
+    Removed deprecated function ``ddtrace.contrib.redis.tracers.py:get_traced_redis_from()`` helper. Use redis patching instead.
```
Comment (majorgreys) [inline releasenotes/notes/update-deprecated-contrib-removal-version-ca920d9e1d91b6e0.yaml:12]:
```suggestion
    Removed deprecated middleware ``ddtrace.contrib.flask.middleware.py:TraceMiddleware``. Use :py:meth:`ddtrace.patch_all` or :py:meth:`ddtrace.patch` instead.
  - |
    Removed deprecated function ``ddtrace.contrib.pymongo.patch.py:trace_mongo_client``. Use :py:meth:`ddtrace.patch_all` or :py:meth:`ddtrace.patch` instead.
    Removed deprecated function ``ddtrace.contrib.pymysql.tracers.py:get_traced_pymysql_connection``. Use :py:meth:`ddtrace.patch_all` or :py:meth:`ddtrace.patch` instead.
  - |
    Removed deprecated function ``ddtrace.contrib.redis.tracers.py:get_traced_redis``. Use :py:meth:`ddtrace.patch_all` or :py:meth:`ddtrace.patch` instead.
  - |
    Removed deprecated function ``ddtrace.contrib.redis.tracers.py:get_traced_redis_from``. Use :py:meth:`ddtrace.patch_all` or :py:meth:`ddtrace.patch` instead.
```

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**